drivers/lis2dh12/lis2dh12.c

```diff
@@ -187,6 +187,55 @@ int lis2dh12_read(const lis2dh12_t *dev, int16_t *data)
     return LIS2DH12_OK;
 }
 
+int lis2dh12_write(const lis2dh12_t *dev, uint8_t reg, uint8_t data)
+{
+    _write(dev, reg, data);
+    return LIS2DH12_OK;
+}
+
+
+/* read interrupt INT_X (INT_1 or INT_2) */
+int lis2dh12_set_interrupt(const lis2dh12_t *dev, int_params_t params, uint8_t INT_X)
+{
+    assert(dev && params.cfg && params.ths && params.duration && params.type);
+
+    assert(INT_X == INT_1 || INT_X == INT_2);
+
+    _acquire(dev);
+    if (INT_X == INT_1){
+        _write(dev, REG_CTRL_REG3, params.type);
+        _write(dev, REG_INT1_CFG, params.cfg);
+        _write(dev, REG_INT1_THS, params.ths);
+        _write(dev, REG_INT1_DURATION, params.duration);
+    }
+    else if (INT_X == INT_2){
+        _write(dev, REG_CTRL_REG6, params.type);
+        _write(dev, REG_INT2_CFG, params.cfg);
+        _write(dev, REG_INT2_THS, params.ths);
+        _write(dev, REG_INT2_DURATION, params.duration);
+    } 
+    _release(dev);
+
+    return LIS2DH12_OK;
+}
+
+/* read interrupt INT_X (INT_1 or INT_2) */
+int lis2dh12_read_interrupt(const lis2dh12_t *dev, uint8_t *data, uint8_t INT_X)
```

Comment:
Reading the code suggests that the function is not ideally named. I would suggest `lis2dh12_read_int_src(dev, src, interrupt_line)` as signature. I would also suggest to make `src` a less opaque type. You could make it an enum or bitfield, perferably an enum (be sure to include a `LIS2DH12_INT_SRC_NONE` (or similar name) default value in the enum).